runtime/cudaq/platform/default/rest_server/helpers/server_impl/CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+include(FetchContent)
+FetchContent_Declare(
+    asio
+    GIT_REPOSITORY    https://github.com/chriskohlhoff/asio.git
```

Comment:
Should this be added to our `tpls` directory as a submodule?

Reply:
This and `crow_cpp` are header-only libs and intended for a very limited scope of this server implementation (behind an interface). I thought using CMake here would be helpful for future portability, e.g., pulling this out of CUDA Quantum in-tree build.

Reply:
OK, that's fair enough. We just need to make sure it is added to our dependency list. @bettinaheim

Reply:
I've added these new dependencies in the NOTICE file: https://github.com/NVIDIA/cuda-quantum/pull/1012/commits/034a3a58295de1d10c1cc1ddc3a979e8e98b7f31

Reply:
The notice file is indeed should contain all dependencies.